eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -89,16 +89,17 @@ sealed trait PaymentRelayed extends PaymentEvent {
   val timestamp: TimestampMilli
 }
 
-case class ChannelPaymentRelayed(amountIn: MilliSatoshi, amountOut: MilliSatoshi, paymentHash: ByteVector32, fromChannelId: ByteVector32, toChannelId: ByteVector32, timestamp: TimestampMilli = TimestampMilli.now()) extends PaymentRelayed
+case class ChannelPaymentRelayed(amountIn: MilliSatoshi, amountOut: MilliSatoshi, paymentHash: ByteVector32, fromChannelId: ByteVector32, toChannelId: ByteVector32, timestampBegin: TimestampMilli, timestamp: TimestampMilli) extends PaymentRelayed
```

Comment:
nit (needs to update the `PaymentRelayed` trait, which should probably contain both timestamps, and we should measure that for trampoline payments as well):
```suggestion
case class ChannelPaymentRelayed(amountIn: MilliSatoshi, amountOut: MilliSatoshi, paymentHash: ByteVector32, fromChannelId: ByteVector32, toChannelId: ByteVector32, relayedAt: TimestampMilli, settledAt: TimestampMilli) extends PaymentRelayed
```

Reply:
You're still using one `timestamp` field, which is quite vague? I think that `PaymentRelayed` should be updated like this:

```
sealed trait PaymentRelayed extends PaymentEvent {
  val amountIn: MilliSatoshi
  val amountOut: MilliSatoshi
  val startedAt: TimestampMilli
  val settledAt: TimestampMilli
}
```

This is clearer for readers, isn't it?

Reply:
The `timestamp` field was part of the supertrait so I can't rename it. I have now duplicated it to give it a new name.

Reply:
Perfect, that's looking better :+1: